### ISSUE TITLE
Is Google's gmock required?

### ISSUE THREAD

**Issue by jeremiah** [Luxoft/Safe-Renderer#2]
Do we have to ship Google's gmock source code? Isn't this just used for testing? If so, can we ask that those who want to test download gmock themselves? This way we would reduce the number of build dependencies.

**Comment by ArwedL** [comment]
You are right: It is included because of testing. Will assign Thomas (system architect) to analyze and give a more solid answer from technical perspective.

**Comment by jeremiah** [comment]
That sounds good, if we can distribute as little third party code as possible that limits our exposure to licensing issues and maintenance.

**Comment by THaarer** [comment]
We maybe could connect the google mock as a git submodule (as gmock is also hosted on github). Then it would be still convenient to build with tests without having the sources in the repository. What do you think?

**Comment by jeremiah** [comment]
Sure! That may be an elegant solution. :-) As I say, I want to limit the amount of code we ship but if you're fine with a submodule that would work for me.

**Comment by denisacostaq** [comment]
Hi @ArwedL & @THaarer & @jeremiah as the test framework is required only if the `UNIT_TESTS` CMake option is `ON`, I recommend you to check the [ExternalProject](https://cmake.org/cmake/help/latest/module/ExternalProject.html) CMake module, this can solve remote content at build time, and of course, only if it is required or outdated.
